code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -61,17 +63,32 @@ var/global/list/parasites = list() //all currently existing/living guardians
 		if("tech")
 			for(var/type in (subtypesof(/datum/guardianname/tech) - namedatum.type))
 				possible_names += new type
+		if("carp")
+			for(var/type in (subtypesof(/datum/guardianname/carp) - namedatum.type))
+				possible_names += new type
 	namedatum = pick(possible_names)
-	updatetheme()
+	updatetheme(pickedtheme)
 
-/mob/living/simple_animal/hostile/guardian/proc/updatetheme() //update the guardian's theme to whatever its datum is; proc for adminfuckery
+/mob/living/simple_animal/hostile/guardian/proc/updatetheme(theme) //update the guardian's theme to whatever its datum is; proc for adminfuckery
 	name = "[namedatum.prefixname] [namedatum.suffixcolour]"
 	real_name = "[name]"
 	icon_living = "[namedatum.parasiteicon]"
 	icon_state = "[namedatum.parasiteicon]"
 	icon_dead = "[namedatum.parasiteicon]"
 	bubble_icon = "[namedatum.bubbleicon]"
 
+	if (namedatum.stainself)
+		color = namedatum.colour
+
+	//Special case holocarp, because #snowflake code
+	if(theme == "carp")
```

Comment:
Now that themes are growing these extra functionalities and becoming quite advance in how they modify the holoparasite I'm saying that now is an excellent time to datumise the themes so you can add special handlings procs to the themes and call them from the holoparasite methods.

That' lets you rework the methods that use if(theme=='type') to be theme.procforthis(variables, variables) and gives the most flexibility for future developers to add more custom abilities/features to themes.